File: scripts/systemApi/python/mea_stimulate.py
from pythonosc.udp_client import UDPClient

from utils import (
    HOST,
    PORT,
    MeaDeviceName,
    StimulateParameter,
    mea_stimulate,
)

if __name__ == "__main__":
    print("Start")
    client = UDPClient(HOST, PORT)
    ch1_stimulate_parameter = StimulateParameter.default_stimulate_parameter()
    ch1_stimulate_parameter.ch_bi_phasic = True
    ch1_stimulate_parameter.ch_enable = True
    ch1_stimulate_parameter.ch_stimulate_channel = 0
    ch1_stimulate_parameter.ch_phase_one_current = 1000
    ch1_stimulate_parameter.ch_phase_two_current = -500
    ch2_stimulate_parameter = StimulateParameter.default_stimulate_parameter()
    ch2_stimulate_parameter.ch_enable = False
    ch2_stimulate_parameter.ch_stimulate_channel = 1
    ch3_stimulate_parameter = StimulateParameter.default_stimulate_parameter()
    ch3_stimulate_parameter.ch_enable = True
    ch3_stimulate_parameter.ch_stimulate_channel = 2
    ch4_stimulate_parameter = StimulateParameter.default_stimulate_parameter()
    ch4_stimulate_parameter.ch_enable = True
    ch4_stimulate_parameter.ch_stimulate_channel = 3
    mea_stimulate(
        client,
        MeaDeviceName.HubA,
        ch1_stimulate_parameter,
        ch2_stimulate_parameter,
        ch3_stimulate_parameter,
        ch4_stimulate_parameter,
    )
    print("End")

File: scripts/systemApi/matlab/mea_stimulate.m
disp('start')
u = udp(Constants.HOST,Constants.PORT);
fopen(u);
ch1StimulateParameter = StimulateParameter.defaultStimulateParameter();
ch1StimulateParameter.chBiphasic = true;
ch1StimulateParameter.chEnable = true;
ch1StimulateParameter.chStimulateChannel = 0;
ch1StimulateParameter.chPhaseOneCurrent = 1000;
ch1StimulateParameter.chPhaseTwoCurrent = -500;
ch2StimulateParameter = StimulateParameter.defaultStimulateParameter();
ch2StimulateParameter.chEnable = true;
ch2StimulateParameter.chStimulateChannel = 1;
ch3StimulateParameter = StimulateParameter.defaultStimulateParameter();
ch3StimulateParameter.chEnable = false;
ch3StimulateParameter.chStimulateChannel = 2;
ch4StimulateParameter = StimulateParameter.defaultStimulateParameter();
ch4StimulateParameter.chEnable = true;
ch4StimulateParameter.chStimulateChannel = 3;
meaStimulate(u,MeaDeviceName.HubA, ch1StimulateParameter,ch2StimulateParameter,ch3StimulateParameter,ch4StimulateParameter);
fclose(u);
disp('end')